bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/cache/ExpiringCache.java

```diff
@@ -51,17 +51,17 @@ public ExpiringCache(long expiry, Supplier<V> action) {
      */
     @Nullable
     public synchronized V getValue() {
-        if (value == null || isExpired()) {
+        if (value.get() == null || isExpired()) {
```

Comment:
Couldn't the soft reference be freed between the twice calls to `get()`?
Shouldn't we use a safe programming style and safe the current reference?

```Java
final V valueCur = value.get();
if (valueCur == null || isExpired()) {
    return refreshValue();
} else {
    retturn valueCur;
}
```

Reply:
Yes we should indeed